### PR TITLE
[SDKS-6713] Virtual Threading

### PR DESCRIPTION
**Pull request by nmayorsplit**

### CODE REVIEW

File: client/src/main/java/io/split/client/utils/SplitExecutorFactory.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    private static ThreadFactory buildThreadFactory(ThreadFactory threadFactory, String name) {
+        if (threadFactory != null) {
+            return new ThreadFactoryBuilder()
+                    .setThreadFactory(threadFactory)
+                    .setDaemon(true)
+                    .setNameFormat(name)
+                    .build();
+        } else {
```
Comment (sanzmauro) [inline client/src/main/java/io/split/client/utils/SplitExecutorFactory.java:42]: same here.

File: client/src/main/java/io/split/client/utils/SplitExecutorFactory.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+public class SplitExecutorFactory {
+
+    public static ScheduledExecutorService buildScheduledExecutorService(ThreadFactory threadFactory, String name, Integer size) {
+        if (threadFactory != null) {
+            return Executors.newScheduledThreadPool(size, new ThreadFactoryBuilder()
+                    .setThreadFactory(threadFactory)
+                    .setDaemon(true)
+                    .setNameFormat(name)
+                    .build());
+        } else {
```
[the remark this answers is not in the document]
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/utils/SplitExecutorFactory.java]: I updated.

File: client/src/main/java/io/split/client/utils/SplitExecutorFactory.java (hunk 1 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/utils/SplitExecutorFactory.java]: I updated.

**Review by mredolatti — APPROVED** [review]
nice work!

**Review by mmelograno — APPROVED** [review]

**Pull request merged**